### PR TITLE
Bug fix/improve test status handling

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/786/
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-windows/66/

File: lib/V8/v8-utils.cpp [hunk 1]
```diff
@@ -2707,16 +2707,25 @@ static void JS_Write(v8::FunctionCallbackInfo<v8::Value> const& args) {
...
-      if (flush) {
-        file.flush();
-        file.sync();
+      if (file.good()) {
+        if (flush) {
+          file.flush();
+          file.sync();
+        }
+        bool good = file.good();
+        file.close();
```
Comment (ObiWahn) [inline lib/V8/v8-utils.cpp]: returns nullptr on fail

**Comment by dothebart** [comment]
Tests green as much as it concerns the code changes.

File: lib/V8/v8-utils.cpp [hunk 2]
```diff
@@ -2707,16 +2707,26 @@ static void JS_Write(v8::FunctionCallbackInfo<v8::Value> const& args) {
 
     std::fstream file;
 
+    errno = 0;
+
+    file.exceptions(std::ifstream::goodbit);
```
Comment (ObiWahn) [inline lib/V8/v8-utils.cpp]: Please add a note that this is supposed to disable exceptions.

**Pull request merged**